File: Programmers/체육복.py
def solution(n, lost, reserve):
    
    lost.sort()
    reserve.sort()
    lost_temp = lost

    x = 0
    while x < len(reserve):
        if reserve[x] in lost_temp:
            lost_temp.remove(reserve[x])
            reserve.remove(reserve[x])
            x -= 1
        x += 1

    for i in range(len(reserve)):
        for j in range(len(lost_temp)):
            if reserve[i] - 1 in lost_temp:
                lost_temp.remove(reserve[i] - 1)
                break
            elif reserve[i] + 1 in lost_temp:
                lost_temp.remove(reserve[i] + 1)
                break
                
    return n - len(lost_temp)

def solution(n, lost, reserve):
    answer = 0
    
    c = [0] * (n+1)     # 0:수업 불가, 1:수업 가능
    for i in range(1, n+1):
        if i not in lost:
            c[i] = 1
        elif i in lost and i in reserve:
            c[i] = 1
            lost.remove(i)
            reserve.remove(i)
        elif i > 0 and i in lost and i-1 not in lost and i-1 in reserve:
            c[i] = 1
            lost.remove(i)
            reserve.remove(i-1)
        elif i < n and i in lost and i+1 not in lost and i+1 in reserve:
            c[i] = 1
            lost.remove(i)
            reserve.remove(i+1)
    answer = c.count(1)
    
    return answer

File: JavaScript/체육복.js
function solution(n, lost, reserve) {
    for (let i = 1; i < n + 1; i++) {
        if (lost.includes(i) && reserve.includes(i)) {
            lost.splice(lost.indexOf(i), 1);
            reserve.splice(reserve.indexOf(i), 1);
        }
    }
    for (let j = 1; j < n + 1; j++) {
        if (lost.includes(j)) {
            if (reserve.includes(j - 1)) {
                lost.splice(lost.indexOf(j), 1);
                reserve.splice(reserve.indexOf(j - 1), 1);
            } else if (reserve.includes(j + 1)) {
                lost.splice(lost.indexOf(j), 1);
                reserve.splice(reserve.indexOf(j + 1), 1);
            }
        }
    }
    return n - lost.length;
}

console.log(solution(5, [1, 5], [2]));
// console.log(solution(5, [2, 4], [4]));
// console.log(solution(3, [3], [1]));